.github/workflows/all-static-checks.yaml

```diff
@@ -61,3 +61,62 @@ jobs:
         run: |
           pip install pyyaml
           python tests/sweep_framework/sweeps_workflow_verification.py
+    name: Static analysis
+  clang-tidy:
+    runs-on: ubuntu-latest
+    container: ghcr.io/tenstorrent/tt-metal/tt-metalium/ubuntu-22.04-amd64:latest
+    permissions:
+      pull-requests: write
+      # OPTIONAL: auto-closing conversations requires the `contents` permission
+      contents: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+        submodules: "recursive"
+    - name: Set safe directory for Git
+      run: git config --global --add safe.directory /__w/tt-metal/tt-metal
+    - name: Fetch base branch
+      run: |
+        git remote add upstream "https://github.com/${{ github.event.pull_request.base.repo.full_name }}"
+        git fetch --no-tags upstream "${{ github.event.pull_request.base.ref }}"
+      continue-on-error: true
+    - name: Install clang-tidy
+      run: |
+        sudo apt-get update
+        sudo DEBIAN_FRONTEND=noninteractive apt-get install -y clang-tidy-17 python3.11 python3.11-venv python3.11-dev
+        sudo ln -s $(which clang-tidy-17) /usr/local/bin/clang-tidy
+      continue-on-error: true
+    - name: Prepare compile_commands.json
+      run: |
+        ARCH_NAME=grayskull cmake -B build -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_DISABLE_PRECOMPILE_HEADERS=ON
+      continue-on-error: true
+    - name: Create results directory
+      run: |
+        mkdir clang-tidy-result
+      continue-on-error: true
+    - name: Analyze
+      run: |
+        git diff -U0 "$(git merge-base HEAD "upstream/${{ github.event.pull_request.base.ref }}")" | clang-tidy-diff-17.py -p1 -path build -export-fixes clang-tidy-result/fixes.yml
+        pwd
+        ls -lart clang-tidy-result
+        cat clang-tidy-result/fixes.yml
+      continue-on-error: true
+    - name: Run clang-tidy-pr-comments action
+      uses: platisd/clang-tidy-pr-comments@v1.5.0
+      with:
+        # The GitHub token (or a personal access token)
+        github_token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
FYI, this might have an unexpected side effect.

> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN, with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs. For example, if a workflow run pushes code using the repository's GITHUB_TOKEN, a new workflow will not run even when the repository contains a workflow configured to run when push events occur.